src/permissions/PermissionCallableAllowedContractNativeTokenRecurringAllowance.sol

```diff
@@ -159,7 +161,11 @@ contract PermissionCallableAllowedContractNativeTokenRecurringAllowance is
     ///
     /// @param permissionHash Hash of the permission.
     /// @param callsSpend Value of native token spent on calls.
+    /// Hmm if I'm an app and can trick a user into calling this function, 
+    /// I can mess up the permissions for any session key? 
```

Comment:
Kind of. I call this out in comment on line 160 above where there is a self-DOS risk. Note that because we are using `msg.sender` as a parameter along with `permissionHash` in storage for recurring allowances, there is no risk of different accounts touching each others' permissions. Even if we use the same hash, the accounting is still separate to protect this case.

Given the recurring nature as well, the worst case is that an app convinces a user to mark the allowance as used up for a permission from another app and wait until it resets in the next cycle. Obviously not ideal, but also not that high severity and with no direct incentive so this would just be a grief. We could easily fix this too by adding this to our blocklist in normal txn popup simulation too.